creator-node/test/contentAccessChecker.test.ts

```diff
@@ -0,0 +1,179 @@
+import { generateSignature } from '../src/apiSigning'
+import { getLibsMock } from './lib/libsMock'
+import { loggerFactory } from './lib/reqMock'
+import assert from 'assert'
+import { getApp } from './lib/app'
+import { checkContentAccess } from '../src/contentAccess/contentAccessChecker'
+
+describe('Test content access', function () {
+  let server: any
+  let libsMock: any
+  let redisMock: any
+  let loggerMock: any
+
+  const dummyDNPrivateKey =
+    '0x3873ed01bfb13621f9301487cc61326580614a5b99f3c33cf39c6f9da3a19cad'
+  const badDNPrivateKey =
+    '0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef'
+  const dummyDNDelegateOwnerWallet =
+    '0x1D9c77BcfBfa66D37390BF2335f0140979a6122B'
+
+  const cid = 'QmcbnrugPPDrRXb5NeYKwPb7HWUj7aN2tXmhgwRfw2pRXo'
+  const signedDataFromDiscoveryNode = {
```

Comment:
sorry i missed this before, but the keys here should look :
`premium_content_id`, `premium_content_type`, `cache`, `timestamp`

that's because we'll eventually have other types of premium content, and so these are the keys DN will send

e.g.
`{ premium_content_id: cid, premium_content_type: 'track', cache: true, timestamp: Date.now() }`

Reply:
content node doesn't care about the type though right?

Reply:
it does. because we may have other content that will be premium. when we say premium content, it's not necessarily limited to DN, so I think let's leave the type in CN too for now.

Reply:
tbh im not sure how it's going to fully play out in the future. but trying to be a bit more generic in the face of this unknown may not hurt. you may end up being right that CN won't need to know about the type. but this was previously discussed and agreed upon, so i'd keep it as that unless you have a strong opinion on it

Reply:
I just don't know what the point of it would be other than appending it to the `req` object which even then it would still be unused. It's also a very easy thing to add in the future. DN can send the data to CN if it wants but CN has no good reason to parse it out of the request

Reply:
sync'd offline on this and going with this code as-is in this PR for now.